kodi_addon_checker/check_addon.py

```diff
@@ -384,3 +392,23 @@ def _check_dependencies(report: Report, addon_path, repo_addons):
             if LooseVersion(available_version) < LooseVersion(required_version) and (required_addon not in ignore):
                 report.add(Record(PROBLEM, "Version mismatch for addon %s. Required: %s, Available: %s "
                                   % (required_addon, required_version, available_version)))
+
+
+def get_addon_name(xml_path):
+    tree = ET.parse(xml_path).getroot()
+    return (tree.get("id"), tree.get("version"))
+
+
+def _check_for_existing_addon(report: Report, addon_xml, repo_addons, branch_name):
+    """Check if addon submitted already exists or not"""
+    addon_xml = os.path.join(addon_xml, "addon.xml")
+    addon_name, addon_version = get_addon_name(addon_xml)
+
+    if addon_name not in repo_addons:
+        report.add(Record(INFORMATION, "This is a new addon"))
+    else:
+        report.add(Record(INFORMATION, "%s addon already exist in branch %s with version %s."
+                          % (addon_name, branch_name, repo_addons[addon_name][0])))
+
+        if repo_addons[addon_name][1] is False:
```

Comment:
Can we write this in a better way? I would guess it's trying to get the broken flag, due to the code below, but it's not very obvious.